src/auth/src/credentials/mds_credential.rs

```diff
@@ -0,0 +1,330 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::credentials::traits::dynamic::Credential;
+use crate::credentials::Result;
+use crate::errors::CredentialError;
+use crate::token::{Token, TokenProvider};
+use async_trait::async_trait;
+use http::header::{HeaderName, HeaderValue, AUTHORIZATION};
+use reqwest::header::HeaderMap;
+use reqwest::Client;
+use std::collections::HashMap;
+use std::env;
+use std::sync::LazyLock;
+
+const METADATA_FLAVOR_VALUE: &str = "Google";
+const METADATA_FLAVOR: &str = "metadata-flavor";
+#[allow(dead_code)] // TODO(#442) - implementation in progress
+static METADATA_ROOT: LazyLock<String> = LazyLock::new(|| {
+    format!(
+        "http://{}/computeMetadata/v1",
+        env::var("GCE_METADATA_HOST").unwrap_or_else(|_| {
+            env::var("GCE_METADATA_ROOT").unwrap_or_else(|_| "metadata.google.internal".to_string())
+        })
+    )
+});
```

Comment:
This function seems complex enough to merit testing. 

I wonder if we need it though, are these environment variables used for anything other than testing?  If not, can we make the configuration-via-environment-variables go away?

If we do need this behavior, then (1) move the closure to a function, (2) test all the interesting cases for the function, not just the one where both values are not set.

Reply:
basically python library uses these env variables to pick up the address of the metadata server, which I assumed can be different from the default value see: https://googleapis.dev/python/google-auth/latest/reference/google.auth.environment_vars.html  and here's the corresponding code: https://github.com/googleapis/google-auth-library-python/blob/main/google/auth/compute_engine/_metadata.py#L45. Given it is just the MVP, I think the better approach would be remove these env variables and add it later when we need them.